packages/vitest/src/node/reporters/base.ts

```diff
@@ -93,121 +95,150 @@ export abstract class BaseReporter implements Reporter {
     }
   }
 
-  onTaskUpdate(packs: TaskResultPack[]): void {
-    for (const pack of packs) {
-      const task = this.ctx.state.idMap.get(pack[0])
-
-      if (task) {
-        this.printTask(task)
-      }
-    }
-  }
-
-  /**
-   * Callback invoked with a single `Task` from `onTaskUpdate`
-   */
-  protected printTask(task: Task): void {
-    if (
-      !('filepath' in task)
-      || !task.result?.state
-      || task.result?.state === 'run'
-      || task.result?.state === 'queued') {
+  protected printTestModule(testModule: TestModule): void {
+    const moduleState = testModule.state()
+    if (moduleState === 'queued' || moduleState === 'pending') {
       return
     }
 
-    const suites = getSuites(task)
-    const allTests = getTests(task)
-    const failed = allTests.filter(t => t.result?.state === 'fail')
-    const skipped = allTests.filter(t => t.mode === 'skip' || t.mode === 'todo')
+    let testsCount = 0
+    let failedCount = 0
+    let skippedCount = 0
 
-    let state = c.dim(`${allTests.length} test${allTests.length > 1 ? 's' : ''}`)
+    // delaying logs to calculate the test stats first
+    // which minimizes the amount of for loops
+    const logs: string[] = []
+    const originalLog = this.log.bind(this)
+    this.log = (msg: string) => logs.push(msg)
```

Comment:
This is quite hacky. Could we wrap the log capturing below in try-catch so that we restore the logger in case something breaks? So that we don't accidentally silence whole logger:

```diff
+ try {
    visit(moduleState, testModule.children)
+ } finally {
    this.log = originalLog
+ }
```